src/x/time/range.go

```diff
@@ -135,6 +135,26 @@ func (r Range) Subtract(other Range) []Range {
 	return res
 }
 
+// IterateForwards iterates through a time range by step size in the
+// forwards direction.
+func (r Range) IterateForwards(stepSize time.Duration, f func(t time.Time) bool) {
```

Comment:
I like the mental semantics of `Iterate` with the direction embedded in the name here over `ForEach` with a flag since nothing really exists here, you're breaking it up by step size. I'll do the rename to remove the `s` though and add the return argument name into the signature